### DISCUSSION TITLE
Windows Push Notifications

### DISCUSSION THREAD

**Discussion by docelmo** [shinyorg/shiny#1336] [Q&A]
So first.  I love you guys.  I think this project is gonna make my life 100000000 times easier considering the firebase libraries blow up on IOS right now.  If I were to use the custom push provider how can I make this work with Windows desktop?  The company I work for love them to death but their IT people are less than accommodating when it comes to things like this.  I have been all over the internet looking for examples on how to do this without WNS.  I am happy to build my own because of my previously mentioned issue.  I stumbled across your project and all I can day is damn where were you 2 years ago...  Hopefully it can help me build something awesome.

**Comment by aritchie** [comment]
I appreciate the kind words and I definitely hope Shiny can help you out.  We've been around the xamarin community for a few years.  

For windows, the push providers isn't the option you need.  That was designed specifically for bolting on top of the native mechanisms (ie. Azure Notification Hubs, Firebase (at least the iOS side), etc).  You really need to implement IPushManager and do all of the lifting yourself.   Shiny.Push does have a generic net7 target, so you'll have access to the type.  Beyond that, I tend to avoid windows myself so I can't provide much help there.
